apps/system/js/base_module.js

```diff
@@ -713,14 +769,27 @@
     },
 
     imports: function() {
-      if (!this.constructor.IMPORTS ||
-          typeof(this.constructor.IMPORTS) == 'undefined' ||
-          this.constructor.IMPORTS.length === 0) {
+      var imports = this.constructor.IMPORTS;
+      if (!imports ||
+          (Array.isArray(imports) && imports.length === 0) ||
+          (!Array.isArray(imports) &&
+            (!imports.default || imports.default.length === 0) &&
+            (!imports[DEVICE_TYPE] || imports[DEVICE_TYPE].length === 0)))
+      {
         return this.__imported();
       }
-      this.debug(this.constructor.IMPORTS);
+
+      if (Array.isArray(imports)) {
+        imports = this.constructor.IMPORTS;
+      } else {
+        var defaulImports = imports.default || [];
+        var deviceImports = imports[DEVICE_TYPE] || [];
+        imports = defaulImports.concat(deviceImports);
+      }
```

Comment:
I guess we can reuse `_convertModulesToArray` here. :P


A little suggestion, BTW: 
can we simplify L772-L788 as follow:

```javascript
var imports = this._convertModulesToArray(this.constructor.IMPORTS || []);
if (imports.length === 0) {
  return this.__imported();
}
```